crates/host/src/module.rs

```diff
@@ -157,15 +226,21 @@ impl PlruCache for SerializedModuleCache {
 }
 
 impl SerializedModuleCache {
-    /// Build a default `SerializedModuleCache` with a `Cranelift` that will be used
-    /// to compile modules for serialization as needed.
-    pub fn default_with_cranelift(cranelift: fn() -> Cranelift) -> Self {
+    /// Build a default `SerializedModuleCache` with a fn to create an `Engine`
+    /// that will be used to compile modules from wasms as needed.
+    pub fn default_with_engine(
```

Comment:
arguably this can be renamed to `new_with_engine` or just `new`

Reply:
Sounds reasonable to me, is this a breaking change anyway? If so I'd go for the rename. Otherwise maybe leave a comment behind to do it later?